tests/chainer_tests/functions_tests/loss_tests/test_absolute_error.py

```diff
@@ -26,74 +26,89 @@
      {'shape': (1, 1)},
      ]
 ))
-class TestAbsoluteError(unittest.TestCase):
+@testing.inject_backend_tests(
+    ['test_forward', 'test_backward', 'test_double_backward'],
+    # CPU
+    [{}]
+    # GPU
+    + testing.product({
+        'use_cuda': [True],
+    })
+    # ChainerX
+    + testing.product({
+        'use_chainerx': [True],
+        'chainerx_device': ['native:0', 'cuda:0'],
+    })
+)
+class TestAbsoluteError(testing.FunctionTestCase):
 
     def setUp(self):
-        self.x0 = numpy.random.uniform(-1, 1, self.shape).astype(self.dtype)
+        if self.dtype == numpy.float16:
+            self.check_forward_options.update({'atol': 1e-3, 'rtol': 1e-3})
+            self.check_backward_options.update({'atol': 5e-2, 'rtol': 5e-2})
+            self.check_double_backward_options.update(
+                {'atol': 5e-2, 'rtol': 5e-2})
+        if self.double_backward_options:
```

Comment:
This check does not seem reliable.
It's not guaranteed that `self.double_backward_options` is empty by default.